crates/client/src/api.rs

```diff
@@ -0,0 +1,31 @@
+use exitfailure::ExitFailure;
+use proto::Rustacean;
+use reqwest::Url;
+use serde::Deserialize;
+#[derive(Deserialize, Debug)]
+pub struct Stars {
+    pub amount: Option<u64>,
+}
+
+impl Stars {
+    pub async fn get_amount() -> Result<Self, ExitFailure> {
+        let url = format!(
+            "https://api.github.com/repos/{owner}/{repo}/stargazers",
+            owner = "rustacean-sh",
+            repo = "rustacean.sh"
+        );
+        let url = Url::parse(&*url)?;
+        let res = reqwest::get(url)
+            .await?
+            .json::<Vec<Rustacean>>()
```

Comment:
Here as generic parameter we should pass the type that deserialzes to the response body.

Which looks like:

```json
[
  {
    "login": "octocat",
    "id": 1,
    "node_id": "MDQ6VXNlcjE=",
    "avatar_url": "https://github.com/images/error/octocat_happy.gif",
    "gravatar_id": "",
    "url": "https://api.github.com/users/octocat",
    "html_url": "https://github.com/octocat",
    "followers_url": "https://api.github.com/users/octocat/followers",
    "following_url": "https://api.github.com/users/octocat/following{/other_user}",
    "gists_url": "https://api.github.com/users/octocat/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/octocat/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/octocat/subscriptions",
    "organizations_url": "https://api.github.com/users/octocat/orgs",
    "repos_url": "https://api.github.com/users/octocat/repos",
    "events_url": "https://api.github.com/users/octocat/events{/privacy}",
    "received_events_url": "https://api.github.com/users/octocat/received_events",
    "type": "User",
    "site_admin": false
  }
]
```

As found in: https://docs.github.com/en/rest/activity/starring?apiVersion=2022-11-28

Reply:
We can deserialize into:
```rust
pub struct Stargazer {
    pub login: String,
    pub id: u64,
    pub node_id: String,
    pub avatar_url: String,
    pub gravatar_id: String,
    pub url: String,
    pub html_url: String,
    pub followers_url: String,
    pub following_url: String,
    pub gists_url: String,
    pub starred_url: String,
    pub subscriptions_url: String,
    pub organizations_url: String,
    pub repos_url: String,
    pub events_url: String,
    pub received_events_url: String,
    pub user_type: String,
    pub user_view_type: String,
    pub site_admin: bool,
}
``` 
Do you agree?

Reply:
SGTM! It would be nice to validate if we can have only relevant items in the final struct as well, so we optimize memory usage.

Reply:
Actually I think we could use https://api.github.com/repos/rustacean-sh/rustacean.sh and pick the `stargazers_count` field from there?

That way we avoid paginating and counting manually.

Reply:
Perfect!